python/pyspark/sql/connect/client.py

```diff
@@ -42,6 +43,125 @@
 logging.basicConfig(level=logging.INFO)
 
 
+class ChannelBuilder:
+    """
+    This is a helper class that is used to create a GRPC channel based on the given
+    connection string per the documentation of Spark Connect.
```

Comment:
For a user-facing API (or developer API), it should better have a complete docs like including `Examples` section.

Also, would need to add `versionadded` directive in the docstring.

Reply:
I added some examples and versionadded to 3.4.0 for now, but we have to revisit this if we pull out the client.